lib/dradis/plugins/appscan/importer.rb

```diff
@@ -0,0 +1,92 @@
+module Dradis::Plugins::Appscan
+  class Importer < Dradis::Plugins::Upload::Importer
+    # The framework will call this function if the user selects this plugin from
+    # the dropdown list and uploads a file.
+    # @returns true if the operation was successful, false otherwise
+    def import(params={})
+      logger.info{ "Parsing Appscan Source output from #{ params[:file] }..." }
+      doc = Nokogiri::XML( File.read(params[:file]) )
+      logger.info{ 'Done.' }
+
+      logger.info{ 'Validating Appscan Source output...' }
+      @assessment_run = doc.xpath('/AssessmentRun').first
+      unless @assessment_run &&
+             @assessment_run.attribute("version") &&
+             @assessment_run.attribute("version").text =~ /^9\./
+        error = {
+          'Title' => 'Invalid file format',
+          'File name' => File.basename(params[:file]),
+          'Description' => "The file you uploaded doesn't seem to be a valid .ozasmt file."
+        }
+        logger.fatal{ error['Description'] }
+        error = error.map{|k,v| "#[%s]#\n%s\n" % [k, v] }.join("\n\n")
+        content_service.create_note text: error
+        return false
+      end
+      logger.info { 'Done.' }
+
+      logger.info { 'Processing report...' }
+      issues = {}
+      node   = content_service.create_node(label: 'Appscan Source')
+
+      asmnt_files = @assessment_run.xpath('Assessment/Assessment/AsmntFile')
+      asmnt_files.each do |asmnt_file|
+        findings = asmnt_file.xpath('Finding')
+        already_processed = []
+        findings.each do |finding|
+          next if finding[:exclude] == "true"
+
+          # we found a .ozasmt example with duplicated data_id's,
+          # we make sure to process unique data_id's per file
+          finding_id = finding[:data_id]
+          next if already_processed.include?(finding_id)
+          already_processed << finding_id
+
+          evidence_data = process_evidence(asmnt_file[:file_id], finding_id)
+          vulnerability = evidence_data[:description]
+
+          # map vulnerability to issue, if the issue doesn't exist yet, create it
+          issue = issues[vulnerability]
+          if issue.nil?
```

Comment:
I think there should be a logger message here saying something like `Adding vulnerability` to say at which part the upload process is at.